#VRML_SIM R2021b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.14148444457626144 0.9552809807826498 0.2596543850927334 2.1817070363121727
  position 3.7449038559955974 2.3295017758334993 -2.558840944043717
}
TexturedBackground {
  texture "factory"
}
TexturedBackgroundLight {
  texture "noon_park_empty"
  luminosity 0
}
RectangleArena {
  floorSize 4 4
}
DEF RobotA Robot {
  translation 0 0.06 0
  rotation 1 0 0 0
  children [
    GPS {
      children [
        DEF gps_sensor_shape Solid {
          translation 0 0.03 0
          rotation 1 0 0 0
          scale 0.5 0.5 0.5
          children [
            DEF gps_position Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
                metalness 0.1
              }
              geometry Cylinder {
                height 0.05
                radius 0.05
              }
            }
          ]
          boundingObject USE gps_position
        }
      ]
      name "gpsSensor"
    }
    DEF camera Camera {
      translation 0 0 0.3
      rotation 0 1 0 3.14
      scale 0.08 0.08 0.08
      children [
        DEF camera Shape {
          appearance PBRAppearance {
          }
          geometry Sphere {
            radius 0.13
          }
        }
      ]
      name "cameraSensor"
      boundingObject USE camera
      physics Physics {
      }
      fieldOfView 1
      width 600
      height 600
      recognition Recognition {
      }
    }
    DEF dis_sensor1 DistanceSensor {
      translation 0.1 0 0.3
      rotation 0 1 0 -1.27
      scale 0.08 0.08 0.08
      children [
        DEF dis_sensor Shape {
          appearance PBRAppearance {
            baseColor 1 0.666667 0.498039
            metalness 0.1
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      name "distanceLeft"
      boundingObject USE dis_sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        1 500 0
        2 1000 0
      ]
    }
    DEF dis_sensor2 DistanceSensor {
      translation -0.1 0 0.3
      rotation 0 1 0 -1.87
      scale 0.08 0.08 0.08
      children [
        DEF dis_sensor Shape {
          appearance PBRAppearance {
            baseColor 1 0.666667 0.498039
            metalness 0.1
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      name "distanceRight"
      boundingObject USE dis_sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        1 500 0
        2 1000 0
      ]
    }
    DEF frontLeft HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.15 -0.001 0.15
      }
      device [
        DEF frontLeftWheel RotationalMotor {
          name "frontLeftWheel"
        }
      ]
      endPoint Solid {
        translation 0.150001 -0.001 0.15
        rotation 0 0 1 1.5699977243029235
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0.333333 0.498039
              metalness 0.3
            }
            geometry Cylinder {
              height 0.02
              radius 0.06
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF WheelPhysics Physics {
          density -1
          mass 0.001
        }
      }
    }
    DEF frontRight HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.15 -0.001 0.15
      }
      device [
        DEF frontRightWheel RotationalMotor {
          name "frontRightWheel"
        }
      ]
      endPoint Solid {
        translation -0.15 -0.001 0.15
        rotation 0 0 1 1.56999772717122
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0.333333 0.498039
              metalness 0.3
            }
            geometry Cylinder {
              height 0.02
              radius 0.06
            }
          }
        ]
        name "solid(2)"
        boundingObject USE Wheel
        physics DEF WheelPhysics Physics {
          density -1
          mass 0.001
        }
      }
    }
    DEF backLeft HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.15 -0.001 -0.15
      }
      device [
        DEF backLeftWheel RotationalMotor {
          name "backLeftWheel"
        }
      ]
      endPoint Solid {
        translation 0.15 -0.001 -0.15
        rotation 0 0 1 1.57
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0.333333 0.498039
              metalness 0.3
            }
            geometry Cylinder {
              height 0.02
              radius 0.06
            }
          }
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics DEF WheelPhysics Physics {
          density -1
          mass 0.001
        }
      }
    }
    DEF backRight HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.15 -0.001 -0.15
      }
      device [
        DEF backRightWheel RotationalMotor {
          name "backRightWheel"
        }
      ]
      endPoint Solid {
        translation -0.15 -0.001 -0.15
        rotation 0 0 1 1.57
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0.333333 0.498039
              metalness 0.3
            }
            geometry Cylinder {
              height 0.02
              radius 0.06
            }
          }
        ]
        name "solid(3)"
        boundingObject USE Wheel
        physics DEF WheelPhysics Physics {
          density -1
          mass 0.001
        }
      }
    }
    DEF RobotBody Shape {
      appearance PBRAppearance {
        baseColor 0 1 1
        metalness 0.5
      }
      geometry Box {
        size 0.3 0.07 0.6
      }
    }
  ]
  boundingObject USE RobotBody
  physics Physics {
    density -1
    mass 5
  }
  controller "controller1"
}
DEF Computer Solid {
  translation -0.68 0 0
  children [
    DesktopComputer {
    }
  ]
  name "solid(1)"
  recognitionColors [
    1 0 0
  ]
}
DEF PottedTree Solid {
  translation -1.42 0 1.52
  children [
    PottedTree {
    }
  ]
  name "solid(2)"
  recognitionColors [
    1 0 0
  ]
}
DEF WoodenPalletStack Solid {
  translation -0.932 0 -1.55
  rotation 0 1 0 1.83
  children [
    WoodenPalletStack {
      palletNumber 2
      palletSize 0.4 0.14 1.2
      palletLathNumber 3
      lateralMisalignment 0
      longitudinalMisalignment 0.5
    }
  ]
  name "solid(3)"
  recognitionColors [
    1 0 0
  ]
}
DEF WoodenBox Solid {
  translation 1.63 0 -1.38
  scale 0.4 0.4 0.4
  children [
    WoodenBox {
    }
  ]
  name "solid(4)"
  recognitionColors [
    1 0 0
  ]
}
DEF Armchair Solid {
  translation 0.109048 0 1.17928
  rotation 0 1 0 1.57
  scale 0.5 0.5 0.5
  children [
    Armchair {
    }
  ]
  name "solid(5)"
  recognitionColors [
    1 0 0
  ]
}
